internal/service/advancedclustertpf/resource.go

```diff
@@ -99,80 +96,230 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
+				patchReq.Paused = nil
+				cluster = r.updateAndWait(ctx, &resumeRequest, diags, &plan)
+				if cluster == nil {
+					return
+				}
+			}
+		}
+		cluster = r.updateAndWait(ctx, patchReq, diags, &plan)
+		if cluster == nil {
+			return
+		}
+		if pauseAfter {
+			cluster = r.updateAndWait(ctx, &pauseRequest, diags, &plan)
 		}
 	}
 	patchReqProcessArgs := update.PatchPayloadTpf(ctx, diags, &state.AdvancedConfiguration, &plan.AdvancedConfiguration, NewAtlasReqAdvancedConfiguration)
 	if patchReqProcessArgs != nil {
-		err := StoreUpdatePayloadProcessArgs(patchReqProcessArgs)
+		advConfig, _, err = api.UpdateClusterAdvancedConfiguration(ctx, projectID, clusterName, patchReqProcessArgs).Execute()
 		if err != nil {
-			diags.AddError("error storing update payload advanced config", fmt.Sprintf("error storing update payload: %s", err.Error()))
+			diags.AddError("errorUpdateAdvancedConfig", fmt.Sprintf(errorConfigUpdate, clusterName, err.Error()))
 		}
 	}
 	patchReqProcessArgsLegacy := update.PatchPayloadTpf(ctx, diags, &state.AdvancedConfiguration, &plan.AdvancedConfiguration, NewAtlasReqAdvancedConfigurationLegacy)
 	if patchReqProcessArgsLegacy != nil {
-		err := StoreUpdatePayloadProcessArgsLegacy(patchReqProcessArgsLegacy)
+		legacyAdvConfig, _, err = r.Client.AtlasV220240530.ClustersApi.UpdateClusterAdvancedConfiguration(ctx, projectID, clusterName, patchReqProcessArgsLegacy).Execute()
 		if err != nil {
-			diags.AddError("error storing update payload advanced config legacy", fmt.Sprintf("error storing update payload: %s", err.Error()))
+			diags.AddError("errorUpdateAdvancedConfigLegacy", fmt.Sprintf(errorConfigUpdate, clusterName, err.Error()))
 		}
 	}
-	if diags.HasError() {
-		return
-	}
-
-	// TODO: Use requests to do actual updates with Admin API
-	tfNewModel, shouldReturn := mockedSDK(ctx, diags, plan.Timeouts)
-	// TODO: keep project_id and name from plan to avoid overwriting for move_state tests. We should probably do the same with the rest of attributes
-	tfNewModel.Name = plan.Name
-	tfNewModel.ProjectID = plan.ProjectID
-	if shouldReturn {
-		return
+	if cluster != nil {
+		r.convertClusterAddAdvConfig(ctx, legacyAdvConfig, advConfig, cluster, plan.Timeouts, diags, &resp.State)
+	} else {
+		r.readCluster(ctx, &plan, &resp.State, diags)
 	}
-	diags.Append(resp.State.Set(ctx, tfNewModel)...)
 }
-
 func (r *rs) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
 	var state TFModel
-	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
-	if resp.Diagnostics.HasError() {
+	diags := &resp.Diagnostics
+	diags.Append(req.State.Get(ctx, &state)...)
+	if diags.HasError() {
 		return
 	}
-	name := state.Name.ValueString()
-	clusterID := state.ClusterID.ValueString()
-	if clusterID == "" {
-		resp.Diagnostics.AddError("errorDelete", fmt.Sprintf(errorDelete, name, "clusterID is empty"))
+	clusterName := state.Name.ValueString()
+	projectID := state.ProjectID.ValueString()
+	if clusterName == "" {
```

Comment:
when can clusterName be empty in Delete? can projectID also be empty?

Reply:
It is required, so shouldn't be empty, think it was a check left from the mocked api

Reply:
i wouldn't do the check then, and let API fails if it's empty